weaver/transform/tiff.py

```diff
@@ -0,0 +1,133 @@
+    :vartype dataset: rasterio.Dataset
+    :ivar is_geotiff: A flag indicating whether the image is a GeoTIFF.
+    :vartype is_geotiff: bool
+    :ivar images: The list of image arrays for multi-page TIFFs.
+    :vartype images: List[np.ndarray]
+    :ivar _images: A copy of the list of image arrays.
+    :vartype _images: List[np.ndarray]
+    :ivar nb_bands: The number of bands in the GeoTIFF.
+    :vartype nb_bands: int
+    :ivar width: The width of the image.
+    :vartype width: int
+    :ivar height: The height of the image.
+    :vartype height: int
+    :ivar bands: A dictionary of band indexes and their data types.
+    :vartype bands: dict
+    :ivar coordinates: The coordinates for the GeoTIFF.
+    :vartype coordinates: Tuple[Tuple[float, float], Tuple[float, float]]
+    :ivar crs: The coordinate reference system for the GeoTIFF.
+    :vartype crs: rasterio.crs.CRS
+    """
+
+    def __init__(self, file_path: str):
+        """
+        Initialize the Tiff object with the given file path.
+
+        :param file_path: The file path to the TIFF image.
+        :type file_path: str
+        """
+        self.file_path = file_path
+        self.dataset = rasterio.open(self.file_path)
+
+        self.is_geotiff = self.dataset.crs is not None
+
+        if not self.is_geotiff:
+            try:
+                self.images = mtif.read_stack(self.file_path)
+                self._images = self.images.copy()
+            except Exception as ex:
+                if isinstance(ex, UnidentifiedImageError):
+                    self.is_geotiff = True
+                else:
+                    raise
+
+        if self.is_geotiff:
+            self.nb_bands = self.dataset.count
+            self.width = self.dataset.width
+            self.height = self.dataset.height
+
+            self.bands = {i: dtype for i, dtype in zip(self.dataset.indexes, self.dataset.dtypes)}
+            self.coordinates = (self.dataset.transform * (0, 0), self.dataset.transform * (self.width, self.height))
+
+            self.crs = self.dataset.crs
+
+    @property
+    def range(self) -> range:
+        """
+        Get the range of valid band indexes for the TIFF file.
+
+        :return: A range object representing valid band indexes.
+        :rtype: range
+        """
+        return range(1, self.nb_bands + 1)
+
+    def get_band(self, index: int) -> Optional[np.ndarray]:
+        """
+        Retrieve a specific band of the image by index.
+
+        :param index: The band index to retrieve.
+        :type index: int
+        :return: The band as a NumPy array, or None if not found.
+        :rtype: Optional[np.ndarray]
+        :raises RuntimeError: If the band index is invalid or data cannot be read.
+        """
+        if index in self.range:
+            return self.dataset.read(index)
+        return None
+
+    def get_images(self, red_band: int = 1, green_band: int = 2, blue_band: int = 3) -> List[Image.Image]:
```

Comment:
Self-note:  "ok" for now, but will most likely make no sense when involving actual satellite geotiffs with other bands (eg: sentinel-2 RGB=(B04,B03,B02)). Maybe `rasterio` could help us directly handling/detecting common band/mission combinations to be handled automatically here...